packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -104,266 +68,583 @@ describe("Garbage Collection Tests", () => {
+            gcSnapshotTree.blobs[`${gcBlobPrefix}_${blob1Id}`] = blob1Id;
+            gcSnapshotTree.blobs[`${gcBlobPrefix}_${blob2Id}`] = blob2Id;
+            gcSnapshotTree.blobs[`${gcBlobPrefix}_${blob3Id}`] = blob3Id;
+
+            // Create a base snapshot that contains the above GC snapshot tree.
+            const baseSnapshot = getDummySnapshotTree();
+            baseSnapshot.trees[gcTreeKey] = gcSnapshotTree;
+
+            const getNodeGCDetails = (blobId: string) => {
+                return gcBlobMap.get(blobId) ?? {};
+            };
+            const garbageCollector = createGarbageCollector(baseSnapshot, getNodeGCDetails);
+
+            // For the nodes in the GC snapshot blobs, remove their references from the default GC data.
+            defaultGCData.gcNodes[nodes[0]] = [];
+            defaultGCData.gcNodes[nodes[1]] = [];
+            defaultGCData.gcNodes[nodes[2]] = [];
+
+            await garbageCollector.collectGarbage({ runGC: true });
+
+            // Change the nodes and validate that an inactiveObjectChanged event is generated for each.
+            garbageCollector.nodeChanged(nodes[1]);
+            garbageCollector.nodeChanged(nodes[2]);
+            garbageCollector.nodeChanged(nodes[3]);
+            assert(
+                mockLogger.matchEvents([
+                    { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[1] },
+                    { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[2] },
+                    { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[3] },
+                ]),
+                "inactiveObjectChanged event not generated as expected",
+            );
+        });
     });
 
-    it("generates inactive events once per node", async () => {
-        const garbageCollector = createGarbageCollector();
+    /**
+     * These tests validate such scenarios where nodes transition from unreferenced -> referenced -> ureferenced state
+     * by verifing that their unreferenced timestamps are updated correctly.
+     *
+     * In these tests, V = nodes and E = edges between nodes. Root nodes that are always referenced are marked as *.
+     */
+     describe("References between summaries - state transition from unreferenced -> referenced -> unreferenced", () => {
+        let garbageCollector: IGarbageCollector;
+        const nodeA = "/A";
+        const nodeB = "/B";
+        const nodeC = "/C";
+        const nodeD = "/D";
+
+        /**
+         * Function that asserts the given test result fails. This is because all the scenarios here currently fail.
+         * These should pass once this issue is fixed - https://github.com/microsoft/FluidFramework/issues/7924. The
+         * assert condition will be flipped then.
+         */
+        function assertTestFails(testResult: boolean, message: string) {
+            assert(!testResult, message);
```

Comment:
Note this function reverts the test result which basically means that the tests don't work as expected now.

Reply:
Please add issue # in the code that tracks it (even if it's next PR :))

Reply:
There is issue# in the comment of this function :-).